.github/workflows/slack-notify.yml

```
@@ -0,0 +1,90 @@
+          PR_NUMBER="${{ github.event.pull_request.number }}"
+          PR_URL="${{ github.event.pull_request.html_url }}"
+          PR_AUTHOR="${{ github.event.pull_request.user.login }}"
+          REPOSITORY="${{ github.repository }}"
+          ACTION_TYPE="${{ github.event.action }}"
+          REVIEWERS_JSON='${{ toJson(github.event.pull_request.requested_reviewers) }}'
+          PR_REVIEWER="${{ github.event.requested_reviewer.login }}"
+          
+          declare -A GITHUB_TO_SLACK
+          GITHUB_TO_SLACK["soeun2537"]="U09A0LM0CRW"
+          GITHUB_TO_SLACK["taek2222"]="U099ARRH3D3"
+          GITHUB_TO_SLACK["changuii"]="U099BR9RNE6"
+          GITHUB_TO_SLACK["eoehd1ek"]="U0995NANDML"
+          GITHUB_TO_SLACK["oungsi2000"]="U098U2R57NK"
+          GITHUB_TO_SLACK["parkjiminnnn"]="U098U8SLXHD"
+          GITHUB_TO_SLACK["etama123"]="U0995MPSZ62"
+          
+          SLACK_AUTHOR_ID=${GITHUB_TO_SLACK[$PR_AUTHOR]:-"$PR_AUTHOR"}
+          SLACK_AUTHOR_MENTION="<@$SLACK_AUTHOR_ID>"
+
+          ############################
+          # 최초 리뷰 요청 알림
+          ############################
+          if [[ "$ACTION_TYPE" == "opened" || "$ACTION_TYPE" == "ready_for_review" ]]; then
+            REVIEWER_MENTIONS=""
+            for reviewer in $(echo "$REVIEWERS_JSON" | jq -r '.[].login'); do
+              slack_id=${GITHUB_TO_SLACK[$reviewer]:-"$reviewer"}
+              REVIEWER_MENTIONS+="<@$slack_id> "
+            done
+
+            if [ -z "$REVIEWER_MENTIONS" ]; then
+              echo "리뷰어가 없으므로 알림 생략"
+              exit 0
+            fi
+
+            curl -X POST -H 'Content-type: application/json' \
+              --data "{\"text\": \"📢 *리뷰 요청 알림*\n*PR 제목:* ${PR_TITLE} (#${PR_NUMBER})\n*작성자:* ${SLACK_AUTHOR_MENTION}\n*리뷰어:* ${REVIEWER_MENTIONS}\n*링크:* ${PR_URL}\"}" \
+              $SLACK_WEBHOOK_URL
+
+            exit 0
+          fi
+          
+          ############################
+          # 재리뷰 요청 알림
+          ############################
+          if [[ "$ACTION_TYPE" == "review_requested" ]]; then
+            slack_id=${GITHUB_TO_SLACK[$PR_REVIEWER]:-"$PR_REVIEWER"}
+            SLACK_REVIEWER_MENTION="<@$slack_id>"
+          
+            REVIEW_API_URL="https://api.github.com/repos/$REPOSITORY/pulls/$PR_NUMBER/reviews"
+            REVIEW_COUNT=$(curl -s -H "Authorization: Bearer $GITHUB_TOKEN" "$REVIEW_API_URL" \
+              | jq -r --arg reviewer "$PR_REVIEWER" '[.[] | select(.user.login == $reviewer)] | length')
+          
+            if [ "$REVIEW_COUNT" -eq 0 ]; then
+              echo "최초 리뷰 요청으로 간주되어 알림 생략"
+              exit 0
+            fi
+        
+            curl -X POST -H 'Content-type: application/json' \
+              --data "{\"text\": \"🏸 *재리뷰 요청 알림*\n*PR 제목:* ${PR_TITLE} (#${PR_NUMBER})\n*작성자:* ${SLACK_AUTHOR_MENTION}\n*리뷰어:* ${SLACK_REVIEWER_MENTION}\n*링크:* ${PR_URL}\"}" \
+              $SLACK_WEBHOOK_URL
+          fi
```

```suggestion

      # Step 1: 최초 리뷰 요청 알림
      - name: Notify initial reviewers
        if: github.event.action == 'opened' || github.event.action == 'ready_for_review'
        run: |
          # 스크립트의 첫 번째 블록 코드를 여기에 붙여넣습니다.
          # REVIEWERS_JSON, GITHUB_TO_SLACK 등의 변수는 환경 변수나 다른 스텝에서 설정해야 합니다.
          # ...
          
      # Step 2: 재리뷰 요청 알림
      - name: Notify re-requested reviewer
        if: github.event.action == 'review_requested'
        run: |
          # 스크립트의 두 번째 블록 코드를 여기에 붙여넣습니다.
          # PR_REVIEWER, GITHUB_TOKEN 등의 변수도 마찬가지로 설정해야 합니다.
          # ...

```

스텝을 분리하면 더 읽기 좋을 것 같아용!

---

반영!